.travis.yml

```diff
@@ -33,6 +33,9 @@ script:
 
 # Cabal
  - cd Cabal
+# Test if gen-extra-source-files.sh was run recently enough
+ - ./misc/gen-extra-source-files.sh
+ - git diff-files
```

Comment:
You need to run `diff-files` with `--exit-code`, otherwise it always returns 0:

```
$ rm LICENSE    
$ git diff-files
:100644 000000 489b8a152a9aba3b4700d6374533590c01c259b3 0000000000000000000000000000000000000000 D	LICENSE
$ echo $?       
0
$ git diff-files --exit-code
:100644 000000 489b8a152a9aba3b4700d6374533590c01c259b3 0000000000000000000000000000000000000000 D	LICENSE
$ echo $?                   
1
```